common/djangoapps/enrollment/views.py

```diff
@@ -451,7 +477,19 @@ def post(self, request):
                     )
                     log.warning(msg)
                     return Response(status=status.HTTP_400_BAD_REQUEST, data={"message": msg})
-                response = api.update_enrollment(username, unicode(course_id), mode=mode, is_active=is_active)
+
+                if len(missing_attrs) > 0:
+                    msg = u"Missing enrollment attributes for provided mode"
+                    log.warning(msg)
```

Comment:
Please include the missing attribute names in the error message so we can diagnose the problem.